primitives/src/xdns.rs

```diff
@@ -33,6 +33,43 @@ pub struct FetchXdnsRecordsResponse<AccountId> {
     pub xdns_records: Vec<XdnsRecord<AccountId>>,
 }
 
+/// A preliminary representation of a xdns_record in the onchain registry.
+#[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct TokenRecord {
+    /// Its token 4b Id
+    pub token_id: [u8; 4],
+
+    /// Link to the gateway token is whitelisted on.
+    pub gateway_id: [u8; 4],
+
+    /// Token properties - decimals, symbol, name
+    pub token_props: TokenSysProps,
+}
+
+/// A preliminary representation of a xdns_record in the onchain registry.
+#[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct GatewayRecord<AccountId> {
```

Comment:
One thing that is kind of missing here is info on the gateway's native asset, so decimals, and ideally also symbol.